packages/utils/src/file/abstract-file-provider.ts

```diff
@@ -0,0 +1,24 @@
+import { FileTypes, IFileProvider } from "@medusajs/types"
+
+export class AbstractFileProviderService implements IFileProvider {
```

Comment:
question: Are we not missing the protected/private upload methods?

Reply:
@olivermrbl We are (I have a comment about it in the http routes), but I want to first implement the providers and see how the protected endpoint should look like. I feel like it should just be a config, not a separate endpoint,but I'll open that discussion once we have all the core pieces in place.

Reply:
Got it, SGTM.